src/Psalm/Internal/Type/SimpleAssertionReconciler.php

```diff
@@ -2651,10 +2650,6 @@ private static function reconcileCallable(
             ) {
                 $callable_types[] = $type;
                 $redundant = false;
-            } elseif ($type instanceof TArray) {
```

Comment:
Shouldn't this create a TCallableKeyedArray now that TCallableArray doesn't exists?

Reply:
Yes, that would make more sense and I don't remember why I made that change to begin with. It doesn't appear to effect any of the tests either way which